scripts/compare_perf.py

```diff
@@ -99,13 +113,14 @@ def main() -> None:
         # or is within a fixed "probably environmental" range
         if latest_time > baseline_time * 1.2 and latest_time - baseline_time > 5.0:
             errors += 1
-            messages.append(f"🚫 Benchmark {i} is too slow: " f"+{perc:.1f}%")
+            messages.append(f"🚫 Benchmark {name} is too slow: " f"+{perc:.1f}%")
         elif rel_dur > 1.1:
             messages.append(
-                f"⚠ Potential non-blocking slowdown in benchmark {i}: " f"+{perc:.1f}%"
+                f"⚠ Potential non-blocking slowdown in benchmark {name}: "
```

Comment:
maybe also give the actualy time, because 6% slower of 2.5s means nothing I think.
There are lots of noises in how each benchmark is run.

Reply:
What do you make of this:
```
[semgrep.bench.zulip.std] 0.923x Baseline: 6.290, Latest: 5.804
[semgrep.bench.dropbox.std] 0.968x Baseline: 1.789, Latest: 1.732
[semgrep.bench.coinbase.std] 0.849x Baseline: 11.051, Latest: 9.378
[semgrep.bench.apache.std] 0.989x Baseline: 11.496, Latest: 11.375
[semgrep.bench.0c34.std] 0.992x Baseline: 1.697, Latest: 1.683
[semgrep.bench.lodash.std] 0.959x Baseline: 6.095, Latest: 5.842
[semgrep.bench.njs-old-dvna.std] 0.888x Baseline: 10.246, Latest: 9.099
[semgrep.bench.DVWA.std] 0.901x Baseline: 8.323, Latest: 7.498
[semgrep.bench.juice-shop.std] 0.947x Baseline: 17.133, Latest: 16.232
[semgrep.bench.Vulnerable-Flask-App.std] 0.913x Baseline: 9.370, Latest: 8.555
[semgrep.bench.pallets.std] 0.938x Baseline: 5.542, Latest: 5.198
[semgrep.bench.socketio.std] 0.954x Baseline: 3.794, Latest: 3.618
[semgrep.bench.coolMenu.std] 0.946x Baseline: 2.254, Latest: 2.133
[semgrep.bench.t00sh.std] 0.849x Baseline: 4.027, Latest: 3.419
[semgrep.bench.grpc.std] 0.964x Baseline: 3.343, Latest: 3.224
Average: 0.932x, Min: 0.849x, Max: 1.000x
Total Baseline: 102.451 s, Latest: 94.790 s
```
semgrep.bench.t00sh.std is 3.4 s vs. 4.0 s. This is huge. Also, all the 15 benchmarks were faster than the baseline. It's extremely unlikely that it's due to chance if the benchmarks use the same binary and all run in the same clean environment. Things that can make the runs not independent include:
* Innocuous differences in the source code that cause huge differences in the performance of the binary. Note that we switched from byte code to native last week.
* There may be some competing jobs running on the GHA instance at the same time.